File: projects/order-creation/app/src/main/java/com/robinfood/app/usecases/getposresolution/IGetPosResolutionUseCase.java
package com.robinfood.app.usecases.getposresolution;

import com.robinfood.core.dtos.transactionrequestdto.TransactionRequestDTO;
import lombok.NonNull;

public interface IGetPosResolutionUseCase {

    /**
     * use case in charge of the resolution by id pos
     *
     * @param token              the authorization token
     * @param transactionRequest transaction request created
     */
    void invoke(
            @NonNull String token,
            @NonNull TransactionRequestDTO transactionRequest
    );
}

File: projects/posv2/app/src/main/kotlin/com/robinfood/app/usecases/getdailyreportevoucher/IGetPosResolutionUseCase.kt
package com.robinfood.app.usecases.getdailyreportevoucher

import com.robinfood.core.dtos.dailyreportvoucher.DailyReportVoucherRequestDTO
import com.robinfood.core.dtos.posresolutionsequence.PosResolutionSequenceDTO

/**
 * Use case Pos Resolution
 */
interface IGetPosResolutionUseCase {

    /**
     * Sends a request to get pos resolution
     * @param token the authentication token
     * @param dailyReportVoucherRequestDTO request for get information
     * @return the pos resolution by PosId
     */
    suspend fun invoke(
        dailyReportVoucherRequestDTO: DailyReportVoucherRequestDTO,
        token: String
    ): PosResolutionSequenceDTO?
}